### DISCUSSION TITLE
Run commands in background during Yazi navigation

### DISCUSSION THREAD

**Discussion by gclarkjr5** [sxyazi/yazi#1959] [1. Q&A]
### What system are you running Yazi on?

macOS

### What terminal are you running Yazi in?

WezTerm

### `yazi --debug` output

```Shell
No debug report.
```

### Describe the question

Hi Yazi community. I have a general question about trying to write a plugin for Yazi. For my specific use case, I have 2 separate ssh config files that I need to switch between (2 separate Github profiles, one laptop) that depends on what directory I'm in. Currently, I handle this using terminal hooks in Nushell. Is there a way to run shell commands in the background based on certain keystrokes? Or if there is any "directory change event" that I can latch on to and run my custom hooks? See below for an attempt I made at this.

### Anything else?

```lua
local gitsync = function()
        -- check if we are in a git repo
        local output = Command("git"):arg("status"):stderr(Command.PIPED):output()
        if output.stderr ~= "" then
          ya.notify({
              title = "gitsync",
              content = "Not in a git directory",
              level = "warn",
              timeout = 5,
          })
        else
          -- prompt user to switch to config_one or config_two
          local cand = ya.which {
            cands = {
              { on = "f", desc = "switch to config_one git config" },
              { on = "n", desc = "switch to config_two git config" }
            }
          }

          if cand == 1 then
            local swap = Command("nu")
              :args({
                "-c",
                "source",
                "~/.custom_nu/.hosts_config.nu;",
                "swap_config",
                "config_two", 
                "config_one"
              })
              :stderr(Command.PIPED):output()

            ya.notify({
                title = "gitsync",
                content = "switched to config_one git config",
                level = "warn",
                timeout = 5,
            })

          elseif cand == 2 then
            local swap = Command("nu")
              :args({
                "-c",
                "source",
                "~/.custom_nu/.hosts_config.nu;",
                "swap_config",
                "config_one", 
                "config_two"
              })
              :stderr(Command.PIPED):output()

            ya.notify({
                title = "gitsync",
                content = "switched to config_two git config",
                level = "warn",
                timeout = 5,
            })
          end
        end

end

return {
    entry = function()
      gitsync()
    end
}
```

### Validations

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment]
> Or if there is any "directory change event" that I can latch on to and run my custom hooks?

If you're looking to execute some custom commands when the Yazi cwd changes, you can subscribe to the `cd` event using `ps.sub("cd", function() end)`. Here's an example: https://github.com/sxyazi/yazi/blob/cb92bb798089dd5ce2b037fb6d2b5991e2153d00/yazi-plugin/preset/plugins/zoxide.lua#L38-L47

Doc of `ps.sub()`: https://yazi-rs.github.io/docs/plugins/utils#ps.sub

Doc of DDS: https://yazi-rs.github.io/docs/dds
